assets/dev/js/editor/components/dynamic-tags/control-behavior.js

```diff
@@ -13,10 +13,30 @@ module.exports = Marionette.Behavior.extend( {
 		}
 	},
 
+	shouldRenderTools() {
+		const hasDefault = this.getOption( 'dynamicSettings' ).default,
+			hasPro = elementor.helpers.hasPro(),
+			hasTags = this.getOption( 'tags' ).length > 0,
+			shouldHavePromotion = ! hasPro || elementor.helpers.hasProAndNotConnected(),
+			shouldHaveSwitcher = hasPro && hasTags;
```

Comment:
```suggestion
		const hasDefault = this.getOption( 'dynamicSettings' ).default,
			isFeatureAvalibleToUser = elementor.helpers.hasPro() && ! elementor.helpers.hasProAndNotConnected(),
		   	hasTags = this.getOption( 'tags' ).length > 0;
		   	
		   	if (hasDefault) {
		   	 	return false;
		   	} 
		   	
		   	return ! isFeatureAvalibleToUser || hasTags
```

Reply:
Done. 
And yes, the calculation shouldn't be here for sure... The whole module shouldn't be here...
But I didn't want to have dependency and didn't want to change the code as much.

Reply:
Done. 
And yes, the calculation shouldn't be here for sure... The whole module shouldn't be here...
But I didn't want to have dependency and didn't want to change the code as much.